x-pack/plugins/index_lifecycle_management/server/routes/api/snapshot_repositories/register_fetch_route.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import { MIN_SEARCHABLE_SNAPSHOT_LICENSE } from '../../../../common/constants';
+import { ListSnapshotReposResponse } from '../../../../common/types';
+
+import { RouteDependencies } from '../../../types';
+import { addBasePath } from '../../../services';
+import { handleEsError } from '../../../shared_imports';
+
+export const registerFetchRoute = ({ router, license }: RouteDependencies) => {
+  router.get(
+    { path: addBasePath('/snapshot_repositories'), validate: false },
+    async (ctx, request, response) => {
+      if (!license.isCurrentLicenseAtLeast(MIN_SEARCHABLE_SNAPSHOT_LICENSE)) {
+        return response.forbidden({
+          body: i18n.translate('xpack.indexLifecycleMgmt.searchSnapshotlicenseCheckErrorMessage', {
+            defaultMessage:
+              'Use of searchable snapshots requires at least an enterprise level license.',
+          }),
+        });
+      }
+
+      try {
+        const esResult = await ctx.core.elasticsearch.client.asCurrentUser.snapshot.getRepository({
+          repository: '*',
+        });
+        const repos: ListSnapshotReposResponse = {
+          repositories: Object.keys(esResult.body),
+        };
+        return response.ok({ body: repos });
+      } catch (e) {
+        // If ES responds with 404 when looking up all snapshots we return an empty array
+        if (e?.statusCode === 404) {
```

Comment:
nit: I created an issue in ES for this error https://github.com/elastic/elasticsearch/issues/65511. The endpoint to get all repos would also work without `repository` parameter.